docs/architecture/adr-076-sign-mode-unified.md

```diff
@@ -0,0 +1,327 @@
+# ADR 075: Sign Mode Unified
+
+## Changelog
+
+* December 2nd 2024: Initial Draft (Zondax AG)
+
+## Status
+
+DRAFT Not Implemented
+
+## Abstract
+
+This ADR introduces `SIGN_MODE_UNIFIED`, a new signing mode for the Cosmos SDK. This mode aims to provide a simpler and more future-proof signing process, focusing on ease of maintenance and an improved user and developer experience. At the core of this new mode is a concept for storing message specifications in a verifiable and efficient manner: *runtime metadata*. This metadata contains the definitions of all data types for each message of each module, down to their underlying primitive types. Signers can then use this metadata to accurately decode and display the intended transaction fields to the user in a human-readable format, avoiding blind signing on any Cosmos-based chain.
+
+This metadata does not need to be stored on the signer app since, by using a chunked and merkleized version of the metadata, chunks representing the data types involved in the messages can be delivered on-demand to the signer, and the root hash of the tree can be computed from those chunks and proofs from the remaining chunks, thus verifying the metadata integrity. On the signer app side, what the user ends up signing is the `metadata root hash + blob`, this means that the signed transaction can be verified by the validator by checking that the metadata root hash in the signed transaction matches the one locally stored in the node.
+
+## Context
+
+## Challenges with Current sign modes
+
+1. **Complex codebase:**
+   Maintaining both sign modes increases the maintenance overhead and the complexity of the codebase.
+
+2. **Inconsistent signing processes across different Cosmos chains:**
+   Not all Cosmos chains use the same sign mode, which creates inconsistencies in the ecosystem.
+
+3. **Blind signing:**
+   When wallets cannot decode a transaction or smart contract call, it leads to blind signing. Blind signing exposes users to potential security risks, as they may unknowingly sign malicious transactions.
+
+4. **Cross-language Types Inconsistency:**
+   Implementing renderers for each language in which a signer would be implemented can lead to inconsistencies in type interpretation.
+
+5. **Upgradeability:**
+   When new types are added to the system, wallets need to synchronize their renderers to match these new types. This process can lead to delays in supporting new types of transactions.
+
+
+## Main components
+
+Key components of `SIGN_MODE_UNIFIED` include:
+
+* *Metadata struct*: A JSON file containing the definitions of all data types for each message of each module, down to primitive types.
+* *Metadata Digest*: A compact representation of the complete metadata set used for transaction construction. When hashed (digest hash), this struct serves as the root of the merkle tree, enabling efficient verification of metadata integrity.
```

Comment:
A "digest" is usually some sort of summary or even the hash which I think is misleading here. What we have here is a serialization of the document above. I think "Metadata Serialization" or "Metadata file" (or similar) would be better.